opensrp-app/src/main/java/org/smartregister/sync/intent/P2pProcessRecordsService.java

```diff
@@ -0,0 +1,122 @@
+
+import org.smartregister.CoreLibrary;
+import org.smartregister.domain.FetchStatus;
+import org.smartregister.domain.db.EventClient;
+import org.smartregister.repository.AllSharedPreferences;
+import org.smartregister.repository.EventClientRepository;
+import org.smartregister.sync.ClientProcessorForJava;
+import org.smartregister.util.Utils;
+
+import java.util.List;
+
+import timber.log.Timber;
+
+/**
+ * Created by Ephraim Kigamba - ekigamba@ona.io on 10/05/2019
+ */
+
+public class P2pProcessRecordsService extends IntentService {
+
+    /**
+     * Creates an IntentService.  Invoked by your subclass's constructor.
+     *
+     * @param name Used to name the worker thread, important only for debugging.
+     */
+    public P2pProcessRecordsService(String name) {
+        super(name);
+    }
+
+    public P2pProcessRecordsService() {
+        super("P2pProcessRecordsService");
+    }
+
+    @Override
+    protected void onHandleIntent(@Nullable Intent intent) {
+        AllSharedPreferences allSharedPreferences = CoreLibrary.getInstance().context().allSharedPreferences();
+
+        if (allSharedPreferences.isPeerToPeerUnprocessedEvents()) {
+            long eventsMaxRowId = allSharedPreferences.getLastPeerToPeerSyncProcessedEvent();
+            EventClientRepository eventClientRepository = CoreLibrary.getInstance().context().getEventClientRepository();
+
+            while (eventsMaxRowId > -1) {
+                EventClientQueryResult eventClientQueryResult = eventClientRepository.fetchEventClientsByRowId(eventsMaxRowId);
+                List<EventClient> eventClientList = eventClientQueryResult.getEventClientList();
+
+                if (eventClientList.size() > 0) {
+                    try {
+                        getClientProcessor().processClient(eventClientList);
+                        int tableMaxRowId = eventClientRepository.getMaxRowId(EventClientRepository.Table.event);
+
+                        if (tableMaxRowId == eventClientQueryResult.maxRowId) {
+                            eventsMaxRowId = -1;
+                            allSharedPreferences.resetLastPeerToPeerSyncProcessedEvent();
+                        } else {
+                            eventsMaxRowId = eventClientQueryResult.maxRowId;
+                            allSharedPreferences.setLastPeerToPeerSyncProcessedEvent(eventClientQueryResult.maxRowId);
+                        }
+
+                        // Profile images do not have a foreign key to the clients and can therefore be saved during the sync.
+                        // They also do not take long to save and therefore happen during sync
+                        Timber.e("Finished processing %s EventClients", String.valueOf(eventClientList.size()));
+                    } catch (Exception e) {
+                        Timber.e(e);
+                    }
+                } else {
+                    allSharedPreferences.resetLastPeerToPeerSyncProcessedEvent();
```

Comment:
shouldn't you break from the while loop here

Reply:
True